test/storage/bwtree_key_test.cpp

```diff
@@ -0,0 +1,1130 @@
+                      std::memcmp(data_A, data_B, key_size) < 0);
+            break;
+          case 2:
+            EXPECT_EQ(CompactIntsFromProjectedRowEq<2>(metadata, *pr_A, *pr_B),
+                      std::memcmp(data_A, data_B, key_size) == 0 && !modified);
+            EXPECT_EQ(CompactIntsFromProjectedRowCmp<2>(metadata, *pr_A, *pr_B),
+                      std::memcmp(data_A, data_B, key_size) < 0);
+            break;
+          case 3:
+            EXPECT_EQ(CompactIntsFromProjectedRowEq<3>(metadata, *pr_A, *pr_B),
+                      std::memcmp(data_A, data_B, key_size) == 0 && !modified);
+            EXPECT_EQ(CompactIntsFromProjectedRowCmp<3>(metadata, *pr_A, *pr_B),
+                      std::memcmp(data_A, data_B, key_size) < 0);
+            break;
+          case 4:
+            EXPECT_EQ(CompactIntsFromProjectedRowEq<4>(metadata, *pr_A, *pr_B),
+                      std::memcmp(data_A, data_B, key_size) == 0 && !modified);
+            EXPECT_EQ(CompactIntsFromProjectedRowCmp<4>(metadata, *pr_A, *pr_B),
+                      std::memcmp(data_A, data_B, key_size) < 0);
+            break;
+          default:
+            throw std::runtime_error("Invalid compact ints key type.");
+        }
+        delete[] data_B;
+      }
+
+      delete[] data_A;
+    }
+
+    delete[] pr_buffer_A;
+    delete[] pr_buffer_B;
+  }
+}
+
+template <uint8_t KeySize, typename CType, typename Random>
+void CompactIntsKeyBasicTest(type::TypeId type_id, Random *const generator) {
+  IndexKeySchema key_schema;
+  const uint8_t num_cols = (sizeof(uint64_t) * KeySize) / sizeof(CType);
+
+  for (uint8_t i = 0; i < num_cols; i++) {
+    key_schema.emplace_back(catalog::indexkeycol_oid_t(i), type_id, false);
+  }
+
+  const IndexMetadata metadata(key_schema);
+  const auto &initializer = metadata.GetProjectedRowInitializer();
+
+  auto *const pr_buffer = common::AllocationUtil::AllocateAligned(initializer.ProjectedRowSize());
+  std::memset(pr_buffer, 0, initializer.ProjectedRowSize());
+  auto *const pr = initializer.InitializeRow(pr_buffer);
+
+  for (uint8_t i = 0; i < num_cols; i++) {
+    pr->AccessForceNotNull(i);
+  }
+
+  CompactIntsKey<KeySize> key1, key2;
+  key1.SetFromProjectedRow(*pr, metadata);
+  key2.SetFromProjectedRow(*pr, metadata);
+
+  EXPECT_TRUE(std::equal_to<CompactIntsKey<KeySize>>()(key1, key2));
+  EXPECT_EQ(std::hash<CompactIntsKey<KeySize>>()(key1), std::hash<CompactIntsKey<KeySize>>()(key2));
+  EXPECT_FALSE(std::less<CompactIntsKey<KeySize>>()(key1, key2));
+
+  *reinterpret_cast<CType *>(pr->AccessForceNotNull(num_cols - 1)) = static_cast<CType>(1);
+  key2.SetFromProjectedRow(*pr, metadata);
+
+  EXPECT_FALSE(std::equal_to<CompactIntsKey<KeySize>>()(key1, key2));
+  EXPECT_NE(std::hash<CompactIntsKey<KeySize>>()(key1), std::hash<CompactIntsKey<KeySize>>()(key2));
+  EXPECT_TRUE(std::less<CompactIntsKey<KeySize>>()(key1, key2));
+
+  delete[] pr_buffer;
+}
+
+// NOLINTNEXTLINE
```

Comment:
Test description?

Reply:
This isn't new code but sure I can come up with one.